### ISSUE TITLE
The cnn2 is not in the model module

### ISSUE THREAD

**Issue by jluo41** [AnthonyMRios/bio-cnn#2]
Hi,

I tried to run the code with `train.py`. But it raised the following error:

For the code here: `from models.cnn2 import CNN`, there is no `cnn2.py` in the `models` folder, only `cnn.py`. You can check the code here: https://github.com/AnthonyMRios/bio-cnn/blob/bc3a57834baf4636eafa50bdba3b5501532b8505/train.py#L60

When I try the `cnn.py`, it doesn't work, too.

Can you give me any suggestions?

Thanks,

Junjie

**Comment by AnthonyMRios** [comment]
It has been a while since I have played with this. But, I believe you can just use "from models.cnn import CNN". I changed the name from cnn2 to cnn.

Note that this was coded in an old version of Theano that is no longer supported. So, there may be issues getting it to run now.

**Comment by jluo41** [comment]
Hi Anthony,

Thanks for your quick reply. We are trying to replicate your code as the baseline model. Sorry to disturb you so much.

I also try to use `cnn.py` directly. But I also notice some other issues.

For example, here the `dropout` is not imported from Theano (maybe? I am not sure). 

https://github.com/AnthonyMRios/bio-cnn/blob/bc3a57834baf4636eafa50bdba3b5501532b8505/models/cnn.py#L68 

Here I find it also hard to know where `self.Wv` and `self.bv` come from.
https://github.com/AnthonyMRios/bio-cnn/blob/bc3a57834baf4636eafa50bdba3b5501532b8505/models/cnn.py#L45

For the `theano` issue. I strongly agree with you. It is old enough and is not long maintained. Maybe it is too hard for us to debug the code with it. I will ask my supervisors to turn to Keras or PyTorch to construct the baseline by following your paper. Thanks so much for your great work!

Best,

Junjie

**Comment by AnthonyMRios** [comment]
Re Dropout: That comes from the utils.py function. Old school theano did not have many preconstructed methods. :) 

Re Wv: I believe you can just comment those out.

Sorry for the messy code. I was trying to clean everything up before uploading here, but probably missed some things.

**Comment by jluo41** [comment]
Hi Anthony,

Thanks so much for your kindly help!

For the dropout, now I got it. It is from `from utils import *`.

For the Wv, I guess to replace Wv and bw with filter_w or filter_b? (I am not sure).

After solving these two issues, eventually, Theano throws other errors. Maybe it is more convenient for us to use PyTorch or Keras and spare the efforts of struggling with Theano.

Anyway, thanks to the tech development, we can use PyTorch or Keras and don't need to build the NNs from the ground with Theano. And thanks again for your quick responses and help!

Best,

Junjie
